### PR TITLE
Issue-47:NRFU6.11: Updated skip enhancement:nrfu_tests/test_system_hardware_temperature

### PR DESCRIPTION
**Pull request by rahulrajolegslab**
# Please include a summary of the changes

* Updated skip enhancement:nrfu_tests/test_system_hardware_temperature.py

# Any specific logic/part of code you need extra attention on

Skip test case with a valid reason and generate both docx and HTML reports correctly.

# Include the Issue number and link

Make sure to link the issue in the github PR UI 

# List/Attach any dependencies/past issues that are required for this change/provide context

# Type of change

- - [ ] Bug fix (non-breaking change which fixes an issue)
- - [ ] New feature (non-breaking change which adds functionality)
- - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- - [ ] This change requires a documentation update
- - [x] Other (please specify)

# Effort required on reviewers end

- - [x] Easy
- - [ ] Medium
- - [ ] Hard 

# How Has This Been Tested?

## Bug fix

    Include before and after snapshots/screenshots/changed logs
    
## New feature

    Ensure current test cases pass and include newer test cases if required
    
# CI pipeline result

- - [ ] Pass
- - [ ] Fail
  
  If it fails, explain the reason and whether or not we should ignore the failure
  
# Verify Documentation Update

    If applicable, ensure documentation has been updated for ReadMe, Getting Started guide, and Style guide
    
# Additional comments

* Skipped reports when the device is vEOS.

[report_2403201102.docx](https://github.com/aristanetworks/vane/files/14662381/report_2403201102.docx)

[report_2403201101.zip](https://github.com/aristanetworks/vane/files/14662384/report_2403201101.zip)

### CODE REVIEW

File: nrfu_tests/test_system_hardware_temperature.py [hunk 1]
```diff
@@ -247,7 +252,12 @@ def test_system_temperature_sensors(self, dut, tests_definitions):
                     tops.actual_output, tops.expected_output, current_threshold_temp_detail
                 )
 
-        except (AssertionError, AttributeError, LookupError, EapiError) as excep:
+        # For BaseException test case is failing instead of skipping it. Hence, adding
+        # specific exception here.
+        except pytest.skip.Exception:
+            pytest.skip(tops.output_msg)
+
+        except (BaseException, EapiError) as excep:
```
Comment (dhruvgoel9) [inline nrfu_tests/test_system_hardware_temperature.py:260]: Revert the changes of except block, don't add pytest skip exception

File: nrfu_tests/test_system_hardware_temperature.py [hunk 2]
```diff
@@ -181,8 +185,9 @@ def test_system_temperature_sensors(self, dut, tests_definitions):
                 f"On device {tops.dut_name}, output of {tops.show_cmd} command is: \n{output}\n",
             )
             self.output += f"\n\nOutput of {tops.show_cmd} command is: \n{output}"
-            power_supplies_sensors = output.get("powerSupplySlots")
-            temperature_sensors = output.get("tempSensors")
+            power_supplies_sensors, temperature_sensors = output.get(
+                "powerSupplySlots"
+            ), output.get("tempSensors")
```
Comment (dhruvgoel9) [inline nrfu_tests/test_system_hardware_temperature.py:190]: Why these changes are needed

File: nrfu_tests/test_system_hardware_temperature.py [hunk 3]
```diff
@@ -169,10 +169,14 @@ def test_system_temperature_sensors(self, dut, tests_definitions):
...
-                pytest.skip(f"{tops.dut_name} is vEOS device, hence test skipped.")
+                tops.output_msg = f"{tops.dut_name} is vEOS device hence the test case was skipped."
+                tests_tools.post_process_skip(
+                    tops, self.test_system_temperature_sensors, self.output
+                )
+                pytest.skip(tops.output_msg)
 
             """
-            TS: Running the "show system environment temperature" command on DUT and
+            TS: Running the `show system environment temperature` command on the device and
```
Comment (dhruvgoel9) [inline nrfu_tests/test_system_hardware_temperature.py:179]: TS changes are not required

**Review by dhruvgoel9 — COMMENTED** [review]
Update the task according to the comments

**Review by dhruvgoel9 — APPROVED** [review]

**Comment by rahulrajolegslab** [comment]
Closing this MR as changes are already handled by MR https://github.com/aristanetworks/vane/pull/703
